### PR TITLE
Service check TTL

### LINKED ISSUE #470: Service check TTL
If TtlScheduler enabled and application have service without heartbeat TtlScheduler task throw exception:

```
2018-11-12 10:46:46.152  INFO 10535 --- [           main] ration$ConsulServiceRegistryFixScheduler : Registering service with consul: NewService{id='core-dbda3022-4cc9-458e-82a4-5e46c5d74b14', name='core', tags=[profile=development, subsystem=core, version=snapshot, secure=false], address='sabo-probook.lan', port=9845, enableTagOverride=null, check=Check{script='null', interval='null', ttl='30s', http='null', tcp='null', timeout='null', deregisterCriticalServiceAfter='120s', tlsSkipVerify=null, status='null'}, checks=null}
2018-11-12 10:46:46.378  INFO 10535 --- [           main] ration$ConsulServiceRegistryFixScheduler : Registering service with consul: NewService{id='core-dbda3022-4cc9-458e-82a4-5e46c5d74b14-management', name='core-management', tags=[management], address='sabo-probook.lan', port=9845, enableTagOverride=null, check=Check{script='null', interval='10s', ttl='null', http='http://sabo-probook.lan:9845/actuator/health', tcp='null', timeout='null', deregisterCriticalServiceAfter='120s', tlsSkipVerify=null, status='null'}, checks=null}

...

2018-11-12 10:46:46.435 ERROR 10535 --- [pool-1-thread-1] o.s.s.s.TaskUtils$LoggingErrorHandler    : Unexpected error occurred in scheduled task.

com.ecwid.consul.v1.OperationException: OperationException(statusCode=500, statusMessage='Internal Server Error', statusContent='CheckID "service:app-dbda3022-4cc9-458e-82a4-5e46c5d74b14-management" does not have associated TTL')
	at com.ecwid.consul.v1.agent.AgentConsulClient.agentCheckPass(AgentConsulClient.java:211)
	at com.ecwid.consul.v1.agent.AgentConsulClient.agentCheckPass(AgentConsulClient.java:198)
	at com.ecwid.consul.v1.agent.AgentConsulClient.agentCheckPass(AgentConsulClient.java:193)
	at com.ecwid.consul.v1.ConsulClient.agentCheckPass(ConsulClient.java:259)
	at org.springframework.cloud.consul.discovery.TtlScheduler$ConsulHeartbeatTask.run(TtlScheduler.java:91)
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:300)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

For example: application service has heartbeat check and application management service has HTTP check.

### PR DESCRIPTION
**Pull request by devpreview**
Closes #470

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@devpreview Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-consul&pullRequestId=471)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-consul&pullRequestId=471) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@devpreview Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-consul&pullRequestId=471)!

**Comment by spencergibb** [comment]
LGTM, can you provide a test?

**Comment by devpreview** [comment]
Test provided.

File: spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/serviceregistry/ConsulServiceRegistryCheckTtlTests.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Alexey Savchuk
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = ConsulServiceRegistryCheckTtlTests.TestConfig.class, properties = {
+        "spring.application.name=myTestService-S",
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/serviceregistry/ConsulServiceRegistryCheckTtlTests.java]: We use tabs, not spaces

**Pull request merged**